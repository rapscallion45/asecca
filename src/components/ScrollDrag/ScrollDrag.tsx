import { FC, useRef, ReactNode, MutableRefObject } from 'react';
import { useDraggable } from 'react-use-draggable-scroll';
import SimpleBarReact from 'simplebar-react';

/**
 * Scroll Drag Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.8
 *
 * @typedef IScrollDragProps
 * @prop {ReactNode} children - component children nodes
 * @prop {any} other - component props to passed through to children
 */
interface IScrollDragProps {
  children: ReactNode;
  other?: any;
}

/**
 * Scroll Drag
 *
 * Global application custom horizontal scroll by mouse drag
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.8
 *
 * @component
 * @param {IScrollDragProps} props - component props
 * @returns {FC} - scroll drag functional component
 */
const ScrollDrag: FC<IScrollDragProps> = (props) => {
  const { children, other } = props;
  const ref = useRef<HTMLDivElement>() as MutableRefObject<HTMLInputElement>;
  const { events } = useDraggable(ref);

  return (
    <SimpleBarReact
      scrollableNodeProps={{ ref }}
      {...events}
      autoHide={false}
      {...other}
    >
      {children}
    </SimpleBarReact>
  );
};

export default ScrollDrag;
