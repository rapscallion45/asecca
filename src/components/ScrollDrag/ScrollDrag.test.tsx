import React from 'react';
import { render } from '@testing-library/react';
import ScrollDrag from './ScrollDrag';

/**
 * Menu Popover Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.22
 */
describe('Menu Popover', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = render(
      <ScrollDrag>
        <div>Test child</div>
      </ScrollDrag>
    );
    expect(tree).toMatchSnapshot();
  });
});
