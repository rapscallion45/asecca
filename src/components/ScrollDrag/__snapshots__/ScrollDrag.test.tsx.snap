// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Popover Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-simplebar="init"
      >
        <div
          class="simplebar-wrapper"
        >
          <div
            class="simplebar-height-auto-observer-wrapper"
          >
            <div
              class="simplebar-height-auto-observer"
            />
          </div>
          <div
            class="simplebar-mask"
          >
            <div
              class="simplebar-offset"
              style="right: 0px; bottom: 0px;"
            >
              <div
                aria-label="scrollable content"
                class="simplebar-content-wrapper"
                role="region"
                style="height: auto; overflow-x: hidden; overflow-y: hidden;"
                tabindex="0"
              >
                <div
                  class="simplebar-content"
                >
                  <div>
                    Test child
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="simplebar-placeholder"
            style="width: 0px; height: 0px;"
          />
        </div>
        <div
          class="simplebar-track simplebar-horizontal"
          style="visibility: hidden;"
        >
          <div
            class="simplebar-scrollbar simplebar-visible"
            style="width: 0px; display: none;"
          />
        </div>
        <div
          class="simplebar-track simplebar-vertical"
          style="visibility: hidden;"
        >
          <div
            class="simplebar-scrollbar simplebar-visible"
            style="height: 0px; display: none;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-simplebar="init"
    >
      <div
        class="simplebar-wrapper"
      >
        <div
          class="simplebar-height-auto-observer-wrapper"
        >
          <div
            class="simplebar-height-auto-observer"
          />
        </div>
        <div
          class="simplebar-mask"
        >
          <div
            class="simplebar-offset"
            style="right: 0px; bottom: 0px;"
          >
            <div
              aria-label="scrollable content"
              class="simplebar-content-wrapper"
              role="region"
              style="height: auto; overflow-x: hidden; overflow-y: hidden;"
              tabindex="0"
            >
              <div
                class="simplebar-content"
              >
                <div>
                  Test child
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="simplebar-placeholder"
          style="width: 0px; height: 0px;"
        />
      </div>
      <div
        class="simplebar-track simplebar-horizontal"
        style="visibility: hidden;"
      >
        <div
          class="simplebar-scrollbar simplebar-visible"
          style="width: 0px; display: none;"
        />
      </div>
      <div
        class="simplebar-track simplebar-vertical"
        style="visibility: hidden;"
      >
        <div
          class="simplebar-scrollbar simplebar-visible"
          style="height: 0px; display: none;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
